FeedStoreChallenge/CoreDataFeedStore.swift

```diff
@@ -0,0 +1,97 @@
+//
+//  CoreDataFeedStore.swift
+//  FeedStoreChallenge
+//
+//  Created by Shilpa Bansal on 23/02/21.
+//  Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+import Foundation
+import CoreData
+
+public enum FeedsEntity: String {
+	case ManagedFeedImage
+	case ManagedCache
+}
+
+public class CoreDataFeedStore: FeedStore {
+	let persistentContainer: NSPersistentContainer
+	let managedContext: NSManagedObjectContext
+	public static let modelName = "FeedStoreDataModel"
+	
+	public init(bundleURL: URL) throws {
+		guard let managedObjectModel = NSManagedObjectModel(contentsOf: bundleURL) else {
+			throw NSError(domain: "Couldnt find the model", code: 0)
+		}
+		
+		persistentContainer = NSPersistentContainer(name: CoreDataFeedStore.modelName, managedObjectModel: managedObjectModel)
+		
+		var loadError: Error?
+		persistentContainer.loadPersistentStores{ loadError = $1 }
+		
+		if let loadError = loadError {
+			throw loadError
+		}
+		managedContext = persistentContainer.newBackgroundContext()
+	}
+	
+	public func deleteCachedFeed(completion: @escaping DeletionCompletion) {
+		perform {[weak managedContext] _ in
+			guard let managedContext = managedContext else {
+				completion(NSError(domain: "Instance not found", code: 0))
+				return
+			}
```

Comment:
This closure receives a `context` as a parameter, but it's being ignored with `_`. It should use the received context instead of capturing the context again as `weak`. So there's no need for the guard statement:

```suggestion
		perform { managedContext in
```